src/isDeepEqual.ts

```diff
@@ -121,6 +121,31 @@ function isDeepEqualImplementation<T, S>(data: S | T, other: S): data is S {
     return data.toString() === (other as unknown as RegExp).toString();
   }
 
+  if (data instanceof Map) {
+    if (data.size !== (other as unknown as Map<unknown, unknown>).size) {
+      return false;
+    }
+
+    const keys = Array.from(data.keys());
+
+    for (const key of keys) {
+      if (!(other as unknown as Map<unknown, unknown>).has(key)) {
+        return false;
+      }
+
+      if (
+        !isDeepEqualImplementation(
+          data.get(key),
+          (other as unknown as Map<unknown, unknown>).get(key),
+        )
+      ) {
+        return false;
+      }
```

Comment:
This is a big implementation, can you pull this logic to a helper function, it would also allow us to cast "other" to a Map<unknown, unknown> once and not have to deal with the type issue on every line.